src/navigation/app/app-navigation.tsx

```diff
@@ -15,17 +15,29 @@ import { useAppSelector, useStackScreenOptions } from 'src/hooks';
 import { AppTabsNavigation } from '../app-tabs';
 import { OnboardingNavigation } from '../onboarding';
 
+// initialRouteName={isUserAdmin ? AppRoute.ADMIN_HOME : AppRoute.APP_TABS}
+
 const Stack = createNativeStackNavigator<AppStackParamList>();
 
-const AppNavigation = () => {
+const AppNavigation: React.FC = () => {
   const { user } = useAppSelector(state => state.auth);
 
+  const isCompletedOnboarding = user?.firstName;
   const isUserAdmin = user?.role === UserRoleType.ADMIN;
 
-  //TODO add check for initial route when Onboarding is needed
+  const initialRouteName = useMemo(() => {
+    if (isUserAdmin) {
+      return AppRoute.ADMIN_HOME;
+    } else if (isCompletedOnboarding) {
+      return AppRoute.APP_TABS;
+    } else {
+      return AppRoute.ONBOARDING_SETUP;
+    }
```

Comment:
Redundant else-if and else statement. Could be simplified

Reply:
```
    if (isUserAdmin) {
      return AppRoute.ADMIN_HOME;
    }
    
    if (isCompletedOnboarding) {
      return AppRoute.APP_TABS;
    }
    
    return AppRoute.ONBOARDING_SETUP;
```